src/ngraph/runtime/cpu/cpu_builder.hpp

```diff
@@ -94,6 +94,24 @@
     else                                                                                           \
         throw ngraph_error("Unsupported rank " + std::to_string(R) + " for kernel " #K);
 
+#define SELECT_RANK2(KV, IT, OT, R, K)                                                             \
+    if (R == 1)                                                                                    \
+        KV = K<IT, OT, 1>;                                                                         \
+    else if (R == 2)                                                                               \
+        KV = K<IT, OT, 2>;                                                                         \
+    else if (R == 3)                                                                               \
+        KV = K<IT, OT, 3>;                                                                         \
+    else if (R == 4)                                                                               \
+        KV = K<IT, OT, 4>;                                                                         \
+    else if (R == 5)                                                                               \
+        KV = K<IT, OT, 5>;                                                                         \
+    else if (R == 6)                                                                               \
+        KV = K<IT, OT, 6>;                                                                         \
+    else if (R == 7)                                                                               \
+        KV = K<IT, OT, 7>;                                                                         \
```

Comment:
out of curiosity; why do we stop at 7?

Reply:
FWIW, TensorFlow seems to stop there for much of its stuff too. I don't really know why. I guess it's a lucky number?